elements/lisk-tree/src/merkle_tree/merkle_tree.ts

```diff
@@ -267,6 +272,134 @@ export class MerkleTree {
+		for (let  i = 0; i < sortedIndexes.length; i += 1) {
+			const index = sortedIndexes[i];
+			if (index.layerIndex === undefined || index.nodeIndex === undefined) {
+				sortedIndexes[i] = sortedIndexes[sortedIndexes.length - 1]
+				sortedIndexes.pop();
+			}
+		}
+		// Indexes must be ordered from bottom-left to top-right
+		(sortedIndexes as NonNullableStruct<NodeIndex>[]).sort((a: NonNullableStruct<NodeIndex>, b: NonNullableStruct<NodeIndex>) => {
+			if (a.layerIndex !== b.layerIndex) return a.layerIndex - b.layerIndex;
+			return a.nodeIndex - b.nodeIndex;
+		});
+
+		const indexes: Record<string, NodeLocation> = {};
+		for (const index of (sortedIndexes as NonNullableStruct<NodeIndex>[])) {
+			const binaryIndex = getBinaryString(index.nodeIndex, this._getHeight() - index.layerIndex).toString();
+			indexes[binaryIndex] = index;
+		}
+
+		const siblingHashes = [];
+		let currentNode: NodeInfo | undefined;
+		let currentNodeHash = Buffer.alloc(0);
+
+		while(!(Object.keys(indexes)[0] === undefined)) {
+			const currentLocation = indexes[Object.keys(indexes)[0]];
+			const {
+				nodeIndex: currentNodeIndex,
+				layerIndex: currentLayerIndex,
+			} = currentLocation;
+			const binaryIndex = getBinaryString(currentNodeIndex, this._getHeight() - currentLayerIndex).toString();
+
+			try {
+				currentNodeHash = (await this._locationToHashMap.get(
+					getBinaryString(currentNodeIndex, this._getHeight() - currentLayerIndex),
+				)) as Buffer;
+				currentNode = await this.getNode(currentNodeHash);
+			} catch (err) {
+				delete indexes[binaryIndex];
+				continue;
+			}
+
+			const {
+				layerIndex: pairLayerIndex,
+				nodeIndex: pairNodeIndex,
+			} = getPairLocation({
+				layerIndex: currentNode.layerIndex,
+				nodeIndex: currentNode.nodeIndex,
+				size: this._size,
+			});
+			const pairNodeLocation: NodeLocation = {
+				layerIndex: pairLayerIndex,
+				nodeIndex: pairNodeIndex,
+			};
+
+			const pairBinaryIndex = getBinaryString(pairNodeIndex, this._getHeight() - pairLayerIndex).toString();
+
+			const pairNodeHash = (await this._locationToHashMap.get(
+				getBinaryString(pairNodeIndex, this._getHeight() - pairLayerIndex),
+			)) as Buffer;
+
+			if (indexes[pairBinaryIndex]) {
+				delete indexes[pairBinaryIndex];
+			} else {
+				siblingHashes.push({
+					hash: pairNodeHash,
+					layerIndex: pairLayerIndex,
+					nodeIndex: pairNodeIndex,
+				});
+			}
+
+			delete indexes[binaryIndex];
+
+			const parentIndex: NodeLocation = getParentLocation(currentLocation, pairNodeLocation);
+			const {
+				layerIndex: parentLayerIndex,
+				nodeIndex: parentNodeIndex,
+			} = parentIndex;
+			const parentBinaryIndex = getBinaryString(parentNodeIndex, this._getHeight() - parentLayerIndex).toString();
+
+			if (parentBinaryIndex !== '0') {
+				indexes[parentBinaryIndex] = parentIndex;
+			}
+		}
+
+		return siblingHashes;
+	}
+
+	public async update(idxs: ReadonlyArray<NodeIndex>, updateData: ReadonlyArray<Buffer>) {
```

Comment:
same here as above. index should be array of number

Reply:
done